### ISSUE TITLE
Training: division by zero in SamplerDataLoader and NaN losses

### ISSUE THREAD

**Issue by avysogorets** [jzhoulab/orca#12]
### Description
Thank you for releasing your code and detailed installation instructions. I followed your exact instructions and tried to reproduce your training workflow by running `train/train_h1esc_a.py` on the default data extracted into the `resources` folder. I also created the `.mmap` object with `python misc/make_genome_memmap.py` prior to running the training script as required. Here are the issues I encounter:

1. I get a bunch of "division by zero" warnings when iterating over the SamplerDartaLoader in line 81 of `train/train_h1esc_a.py`:

`...cooltools/lib/numutils.py:1376: RuntimeWarning: invalid value encountered in divide
  val_cur = ar_cur / armask_cur`

The warning is triggered in the `adaptive_coarsegrain` method in [cooltools.lib.numutils.py](https://cooltools.readthedocs.io/en/latest/cooltools.lib.html#cooltools.lib.numutils.adaptive_coarsegrain).  

2. The second issue might be related to the first one. During training, I observe an exploding `loss_1d`, which results in NaN values in the model output after 2-3 minibatches.

### Ways to reproduce
Just follow the installation instructions in the README files in `orca/` and `orca/train`. Then run `train/train_h1esc_a.py`. My system parameters:
 - Linux Mint 22 Wilma
 - Anaconda: conda 24.9.2
 - Python: python 3.12.7

**Comment by jzthree** [comment]
The first issue is due to cooltools and it should not affect normal training.

For the second, are you training with a different batch size? You might need to lower learning rate of lower batch size is used.

Hope this helps!

**Comment by avysogorets** [comment]
Thank you for your prompt reply. I did lower the batch size from 16 to 8 but I still observe the same behavior however small learning rates I try. The output has NaN values within 3-4 batches and sometimes even on the first batch. The lowest learning rate I tried was 1e-6 when the default is 0.002. I also tried gradient accumulation to effectively restore the original batch size but it didn't work either.

Did you encounter any of these issues when training, including persistent warnings originating from cooltools?
